Sources/BuilderIO/Helpers/RenderContent.swift

```diff
@@ -3,36 +3,52 @@ import SwiftUI
 @available(iOS 15.0, macOS 10.15, *)
 public struct RenderContent: View {
     static var registered = false;
-    
-    public init(content: BuilderContent) {
+    var content: BuilderContent;
+    var apiKey: String;
+
+    public init(content: BuilderContent, apiKey: String) {
         self.content = content
+        self.apiKey = apiKey
         if (!RenderContent.registered) {
             // TODO: move these out of here?
-            registerComponent(name: "Text", factory: { (options, styles) in
+            registerComponent(component: BuilderCustomComponent(name: "Text", inputs: [
+                BuilderInput(name: "text", type: "text")
+            ]), factory: { (options, styles, _) in
                 return BuilderText(text: options["text"].stringValue, responsiveStyles: styles)
-            })
-            registerComponent(name: "Image", factory: { (options, styles) in
-                return BuilderImage(image: options["image"].stringValue, backgroundSize: options["backgroundSize"].stringValue, aspectRatio: CSSStyleUtil.getFloatValue(cssString: options["aspectRatio"].stringValue),  responsiveStyles: styles)
-            })
-            registerComponent(name: "Core:Button", factory: { (options, styles) in
+            }, apiKey: nil)
```

Comment:
instead of nil, should we just pass the apiKey value?

Reply:
passing the apiKey as nil wouldn't send the API request for remote session which is not needed for those internal components